xclim/indices/_conversion.py

```diff
@@ -1180,3 +1173,446 @@ def potential_evapotranspiration(
+    pa = convert_units_to(e_sat, "kPa") * convert_units_to(hurs, "1")
+
+    utci = _utci(tas, sfcWind, delta, pa)
+
+    utci = utci.assign_attrs({"units": "degC"})
+    if mask_invalid:
+        utci = utci.where(
+            (-50.0 < tas)
+            & (tas < 50.0)
+            & (-30 < delta)
+            & (delta < 30)
+            & (0.5 < sfcWind)
+            & (sfcWind < 17.0)
+        )
+    return utci
+
+
+def _fdir_ratio(dates, csza_i, csza_s, rsds):
+    """Return ratio of direct solar radiation."""
+    d = distance_from_sun(dates)
+    s_star = rsds * ((1367 * csza_s * (d ** (-2))) ** (-1))
+    s_star = xr.where(s_star > 0.85, 0.85, s_star)
+    fdir_ratio = np.exp(3 - 1.34 * s_star - 1.65 * (s_star ** (-1)))
+    fdir_ratio = xr.where(fdir_ratio > 0.9, 0.9, fdir_ratio)
+    return xr.where(
+        (fdir_ratio <= 0) | (csza_i <= np.cos(89.5 / 180 * np.pi)) | (rsds <= 0),
+        0,
+        fdir_ratio,
+    )
+
+
+@declare_units(
+    rsds="[radiation]", rsus="[radiation]", rlds="[radiation]", rlus="[radiation]"
+)
+def mean_radiant_temperature(
+    rsds: xr.DataArray,
+    rsus: xr.DataArray,
+    rlds: xr.DataArray,
+    rlus: xr.DataArray,
+    stat: str = "sunlit",
+) -> xr.DataArray:
+    r"""Mean radiant temperature.
+
+    The mean radiant temperature is the incidence of radiation on the body from all directions.
+
+    Parameters
+    ----------
+    rsds : xr.DataArray
+       Surface Downwelling Shortwave Radiation
+    rsus : xr.DataArray
+        Surface Upwelling Shortwave Radiation
+    rlds : xr.DataArray
+        Surface Downwelling Longwave Radiation
+    rlus : xr.DataArray
+        Surface Upwelling Longwave Radiation
+    stat  : {'sunlit', 'instant'}
+        Which statistic to apply. If "sunlit", the cosine of the solar zenith angle
+        is calculated during the sunlit period of each interval. If "instant" the
+        instantaneous cosine of the solar zenith angle is calculated.
+
+    Returns
+    -------
+    xarray.DataArray
+        Mean radiant temperature
+
+    Notes
+    -----
+    This code was inspired by the `thermofeel` package.
+
+    References
+    ----------
+    Di Napoli, C., Hogan, R.J. & Pappenberger, F. Mean radiant temperature from global-scale
+    numerical weather prediction models. Int J Biometeorol 64, 1233–1245 (2020).
+    https://doi.org/10.1007/s00484-020-01900-5
+    """
+    rsds = convert_units_to(rsds, "W m-2")
+    rsus = convert_units_to(rsus, "W m-2")
+    rlds = convert_units_to(rlds, "W m-2")
+    rlus = convert_units_to(rlus, "W m-2")
+
+    dates = rsds.time
+    hours = dates.dt.hour
+    lat = rsds.lat
+    lon = rsds.lon
+    decimal_year = datetime_to_decimal_year(times=dates, calendar=dates.dt.calendar)
+    day_angle = ((decimal_year % 1) * 2 * np.pi).assign_attrs(units="rad")
+    dec = solar_declination(day_angle)
+    if stat == "sunlit":
+        interval = (
+            (dates[1] - dates[0]).values.astype("timedelta64[h]").astype(np.int32)
```

Comment:
```suggestion
            (dates.diff('time').dt.seconds / 3600).reindex(time=dates.time, method='bfill')
```
How's that? One for each timestep and a resolution of 1s.   It assumes the coordinate marks the end of the interval, which is different from the assumption in `;csza` (there it takes the hour as the middle of the interval).